src/coprocessor/dag/executor/mod.rs

```diff
@@ -135,10 +135,16 @@ pub trait Executor {
     fn collect_output_counts(&mut self, counts: &mut Vec<i64>);
     fn collect_statistics_into(&mut self, stats: &mut Statistics);
     fn collect_metrics_into(&mut self, metrics: &mut ScanCounter);
+    /// Take the last key accessed by the executor.
+    ///
+    /// If the executor doesn't support this, return None.
+    fn take_last_key(&mut self) -> Option<Vec<u8>> {
```

Comment:
I think for selectio/limit executor, it should return ` self.src.take_last_key()`  instead of `None`

Reply:
After talk with @tiancaiamao , we don't need to fill the `key_range` field of `Response` for other `Executor`s beside `TableScan` and `IndexScan`. I think now we can comment here.